src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/credential.py

```diff
@@ -14,21 +15,25 @@ def acr_credential_show(registry_name, resource_group_name=None):
     :param str registry_name: The name of container registry
     :param str resource_group_name: The name of resource group
     '''
-    if resource_group_name is None:
-        resource_group_name = get_resource_group_name_by_registry_name(registry_name)
-
+    registry, resource_group_name = get_registry_by_name(registry_name, resource_group_name)
     client = get_acr_service_client().registries
 
-    return client.get_credentials(resource_group_name, registry_name)
+    if registry.admin_user_enabled: #pylint: disable=no-member
+        return client.get_credentials(resource_group_name, registry_name)
+
+    raise CLIError("Run 'az acr update -n {} --admin-enabled true' to enable admin first.".format(
+        registry_name))
 
 def acr_credential_renew(registry_name, resource_group_name=None):
     '''Regenerates the administrator login credentials for the specified container registry.
     :param str registry_name: The name of container registry
     :param str resource_group_name: The name of resource group
     '''
-    if resource_group_name is None:
-        resource_group_name = get_resource_group_name_by_registry_name(registry_name)
-
+    registry, resource_group_name = get_registry_by_name(registry_name, resource_group_name)
     client = get_acr_service_client().registries
 
-    return client.regenerate_credentials(resource_group_name, registry_name)
+    if registry.admin_user_enabled: #pylint: disable=no-member
+        return client.regenerate_credentials(resource_group_name, registry_name)
+
+    raise CLIError("Run 'az acr update -n {} --admin-enabled true' to enable admin first.".format(
```

Comment:
You might extract this out to a variable to avoid duplication.
```Python
admin_not_enabled_error = CLIError('blahblah')
if cats:
  raise admin_not_enabled_error

if dogs:
  raise admin_not_enabled_error
```
That way if you ever have to change it you won't have to remember it's in two places.